### PR TITLE
#5239 set some more mandatory fields

### LINKED ISSUE #5239: Set some more fields as mandatory
## Describe the solution you'd like

To create a patient, we have already set some fields as mandatory. 
Without entering data to mandatory fields, user can't create new patient.

Now, we are going to expand this and setting more fields as mandatory as described in the [excel](https://docs.google.com/spreadsheets/d/1TxGC30J7qJOqOQz_L2L62BTt8URho4-1JS4NS_g_RFM/edit?pli=1#gid=1122312565)

For patient details except formSchema side, we needs a store preference to set more mandatory fields.
mSupply desktop issue: https://github.com/openmsupply/msupply/issues/13386
 
## Implementation

**A description of how the solution would best be implemented in the code base**

## Describe alternatives you've considered

**A clear and concise description of any alternative solutions or features you've considered.**

## Additional context

**Add any other context or screenshots about the feature request here.**

### PR DESCRIPTION
**Pull request by arjunSussol**
Set some more mandatory fields for patient and prescriber.

Fixes #5239 

## Change summary

Explain and justify your changes

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Better to set up mobile with Kiribati data
- [ ] On desktop, open record browser > go to `form_schema` table and query by type `PatientSurvey`. Its json_schema and ui_schema fields needs to be updated
- [ ] For json_schema update use the text from: 
[KiriJsonSchema.txt](https://github.com/openmsupply/mobile/files/12068452/KiriJsonSchema.txt)
- [ ] For ui_schema update use the text from: 
[KiriUISchema.txt](https://github.com/openmsupply/mobile/files/12068471/KiriUISchema.txt)
- [ ] If you still struggle with the schema update then please let me know, I will help you.
- [ ] The required fields that set mandatory are listed here: https://docs.google.com/spreadsheets/d/1TxGC30J7qJOqOQz_L2L62BTt8URho4-1JS4NS_g_RFM/edit?pli=1#gid=1122312565
- [ ] On mobile, you must give or select a valid value for above fields to create/update the patient/prescriber.
- [ ] After approval and merger of this PR, I will update those schema to our [private repo](https://github.com/sussol/client-JSON-schemas/tree/main/Kiribati/COVID-19-vaccination-rollout)

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

### CODE REVIEW

**Comment by arjunSussol** [comment]
[(4.1.4.b) Prevent user from omitting mandatory information](https://trello.com/c/KhoHdBO9/109-414b-prevent-user-from-omitting-mandatory-information)

**Comment by AnushaUp** [comment]
## Testing
Tested in `v8.6.6-rc3` Test APK
Steps to reproduce or otherwise test the changes of this PR:

- [ ] Better to set up mobile with Kiribati data
- [ ] On desktop, open record browser > go to `form_schema` table and query by type `PatientSurvey`. Its json_schema and ui_schema fields needs to be updated
- [ ] For json_schema update use the text from: 
[KiriJsonSchema.txt](https://github.com/openmsupply/mobile/files/12068452/KiriJsonSchema.txt)
- [ ] For ui_schema update use the text from: 
[KiriUISchema.txt](https://github.com/openmsupply/mobile/files/12068471/KiriUISchema.txt)
- [ ] If you still struggle with the schema update then please let me know, I will help you.
- [ ] The required fields that set mandatory are listed here: https://docs.google.com/spreadsheets/d/1TxGC30J7qJOqOQz_L2L62BTt8URho4-1JS4NS_g_RFM/edit?pli=1#gid=1122312565
- [ ] On mobile, you must give or select a valid value for above fields to create/update the patient/prescriber.
- [ ] After approval and merger of this PR, I will update those schema to our [private repo](https://github.com/sussol/client-JSON-schemas/tree/main/Kiribati/COVID-19-vaccination-rollout)

**Comment by AnushaUp** [comment]
> * On mobile, you must give or select a valid value for above fields to create/update the patient/prescriber.
Its creating problems for dispensing vaccine for the patients from other store. It does not let you move forward if the mandatory fields are empty and can't let you input in the fields also.

**Comment by sanjeevsussol** [comment]
@anildahalsussol - can you do the review?

File: src/utilities/formInputConfigs.js [hunk 1]
```diff
@@ -141,12 +141,15 @@ const FORM_INPUT_CONFIGS = seedObject => ({
...
-    initialValue: false,
+    initialValue: null,
     key: 'female',
     options: [true, false],
     optionLabels: [formInputStrings.female, formInputStrings.male],
     label: formInputStrings.gender,
     isEditable: true,
+    isRequired: true,
+    validator: input => input !== null,
+    invalidMessage: `${formInputStrings.gender_must_be_selected}`,
```
Comment (anildahalsussol) [inline src/utilities/formInputConfigs.js:152]: @arjunSussol : Me and @AnushaUp tested and saw this msg is not showing up.

**Review by anildahalsussol — CHANGES_REQUESTED** [review]
Tested with @AnushaUp : Apart from that, we got some flaw regarding json schema like 
Previous serious allergic reactions?: Not shown as required in the UI and also it has default value. Similarly for Previous adverse events following immunisation

**Comment by anildahalsussol** [comment]
> > * On mobile, you must give or select a valid value for above fields to create/update the patient/prescriber.
> 
> Its creating problems for dispensing vaccine for the patients from other store. It does not let you move forward if the mandatory fields are empty and can't let you input in the fields also.

@adamdewey @arjunSussol: Are we okay with this one?

**Comment by adamdewey** [comment]
> @adamdewey @arjunSussol: Are we okay with this one?

No, we can't release like this unfortunately. 

I was talking to @DhanyaHerath earlier - we were thinking that we could put these changes behind a global preference - so that only the K client could use them. Would that be an option?

**Comment by arjunSussol** [comment]
> > @adamdewey @arjunSussol: Are we okay with this one?
> 
> No, we can't release like this unfortunately.
> 
> I was talking to @DhanyaHerath earlier - we were thinking that we could put these changes behind a global preference - so that only the K client could use them. Would that be an option?

Hi @adamdewey even if we add the store preference as you mentioned then still the issue will be raised for the Kiribati sites. One can't dispense vaccine to patient of another store.

**Comment by DhanyaHerath** [comment]
@arjunSussol Oh, I see what you mean, though catch me up a little - does this get checked every time you dispense? Not when you are creating the patient itself?

**Comment by arjunSussol** [comment]
Hi @DhanyaHerath yes, it will be checked every time when dispensed to a patient from different store.

**Comment by arjunSussol** [comment]
> Tested with @AnushaUp : Apart from that, we got some flaw regarding json schema like Previous serious allergic reactions?: Not shown as required in the UI and also it has default value. Similarly for Previous adverse events following immunisation

@anildahalsussol @AnushaUp Have you updated the form_schema as described in the PR description?

**Comment by AnushaUp** [comment]
> > Tested with @AnushaUp : Apart from that, we got some flaw regarding json schema like Previous serious allergic reactions?: Not shown as required in the UI and also it has default value. Similarly for Previous adverse events following immunisation
> 
> @anildahalsussol @AnushaUp Have you updated the form_schema as described in the PR description?

@anildahalsussol @arjunSussol Yeah it was due to the json field not updated properly😅

**Comment by AnushaUp** [comment]
Moving this to done as per the discussion

**Comment by arjunSussol** [comment]
> > @adamdewey @arjunSussol: Are we okay with this one?
> 
> No, we can't release like this unfortunately.
> 
> I was talking to @DhanyaHerath earlier - we were thinking that we could put these changes behind a global preference - so that only the K client could use them. Would that be an option?

@adamdewey @DhanyaHerath I have set the preference in the mobile side, by default it will be `False`.
We can create footrunner for Kiribati and set it to `True`, then validation will work accordingly.
Fixed by https://github.com/openmsupply/mobile/pull/5266/commits/74eaa9f3053bb627a80f71cf42af7dd55597689f

@anildahalsussol @AnushaUp I will create a release version once approved this PR.
Please, make sure that you have updated the `form_schema` before testing this.
Also, we may need a foot-runner for Kiribati data.

**Comment by JonnaQuismundo** [comment]
Hi @anildahalsussol can you re-review? Thanks!

**Review by anildahalsussol — APPROVED** [review]
Looks good. Will test once apk is built.

**Pull request merged**